### DISCUSSION TITLE
tutorials/intro-hotchocolate/03-hot-chocolate

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#146] [course feedback]
# tutorials/intro-hotchocolate/03-hot-chocolate

Learn about the GraphQL server C# / .NET framework, Hot Chocolate.

https://www.apollographql.com/tutorials/intro-hotchocolate/03-hot-chocolate

**Comment by naskonaskov** [comment]
Is there expected release date for the GraphQL .NET course?

**Reply by mabuyo** [reply]
Not at the moment @naskonaskov - but always good to hear when people are interested!

**Comment by ajencydigital** [comment]
Hello, loving the course so far, the way it's explained and the way you guys help us visualize what we're learning is by far the best approach. 

I only have one question: is there a specific reason, or benefit over the others, that made you choose the annotations approach?

Wondering specially because of what you guys just mentioned about how in the end HotChocolate still, regardless of the approach, would transform it to code first anyway. So at that point my guess was you guys were gonna go with code first, so I was surprised when you picked annotations.

**Reply by mabuyo** [reply]
Hey there @ajencydigital - thanks for the feedback. We believed that as a beginner to GraphQL (which is who this course is for), it's easier to get started using annotations! It required less code and looks cleaner. Their documentation does show code snippets for all three approaches though, so really it's up to you (or your team!) to decide.

**Reply by knd775** [reply]
Using annotations *is* code-first

**Reply by mabuyo** [reply]
Hey there @knd775! We are using Hot Chocolate's definitions specifically (note that they've since renamed "annotation-based" to "implementation-first" in v14), and they differentiate between the annotations approach and the more explicit "code-first" approach.

**Comment by lars-thurman** [comment]
It would be helpful for you to tell us what template you are using to create the sample above.
